src/content-tags-drawer/ContentTagsDropDownSelector.jsx

```diff
@@ -22,21 +44,34 @@ const ContentTagsDropDownSelector = ({
   // This object represents the states of the dropdowns on this level
   // The keys represent the index of the dropdown with
   // the value true (open) false (closed)
-  const [dropdownStates, setDropdownStates] = useState({});
+  const [dropdownStates, setDropdownStates] = useState(/** type Record<string, boolean> */ {});
   const isOpen = (tagValue) => dropdownStates[tagValue];
 
   const [numPages, setNumPages] = useState(1);
   const parentTagValue = lineage.length ? decodeURIComponent(lineage[lineage.length - 1]) : null;
   const { hasMorePages, tagPages } = useTaxonomyTagsData(taxonomyId, parentTagValue, numPages, searchTerm);
 
-  const [prevSearchTerm, setPrevSearchTerm] = useState(searchTerm);
-
   // Reset the page and tags state when search term changes
-  // and store search term to compare
-  if (prevSearchTerm !== searchTerm) {
-    setPrevSearchTerm(searchTerm);
+  useEffect(() => {
     setNumPages(1);
-  }
+  }, [searchTerm]);
```

Comment:
This hook will be called every time `searchTerm` changes.

Reply:
Thank you for the docs! I didn't know that this was the recommended way.
Reverted: 62022b5eb309f973d3f9f16cac09b2f30f5d2781